lib/hotel.rb

```diff
@@ -0,0 +1,184 @@
+			end
+		end
+		
+		#makes new reservation
+		new_reservation = Hotels::Reservation.new(room_number, date)
+		@reservations_by_room[room_number] << new_reservation
+		
+		#if this day already has a reservation
+		if @reservations_by_date.key?(date.start_date)
+			#add it to the array
+			@reservations_by_date[date.start_date] << new_reservation
+		else
+			#other wise make a new array and add it
+			@reservations_by_date[date.start_date] = [new_reservation]
+		end
+			
+	  end
+	  
+	  
+	  def get_all_rooms
+		return @rooms
+	  end
+	  
+	  def get_open_rooms(date)
+		open_rooms = []
+		#makes a boolean array flagging all rooms as available
+		20.times do |i|
+			open_rooms[i] = true
+		end
+		
+		current_date = date.start_date
+		while current_date < date.end_date do
+			#if the current_date has a reservation
+			if @reservations_by_date.key?(current_date)
+				current_reservations = @reservations_by_date[current_date]
+				#iterate over each reservation
+				current_reservations.each do  |this_reservation|
+					#set value of the boolean array (where the index == room number) to false
+					open_rooms[this_reservation.room_number] = false
+				end
+			end
+			current_date += 1
+		end
+		
+		#this will hold the actual avaialble rooms
+		available_rooms = []
+		index = 0
+		
+		open_rooms.each do |value|
+			#if the value of the boolean array is true
+			if value == true
+				#append the room at the index to the available_rooms array
+				available_rooms << @rooms[index]
+			end
+			index += 1
+		end
+		
+		return available_rooms
+	  end
+	  
+	  def make_new_block(number_of_rooms, date, id)
+		available_rooms = get_open_rooms(date)
+		if number_of_rooms > 5
+			raise ArgumentException, "Room blocks must be 5 rooms or less"
+		elsif number_of_rooms > available_rooms.length
+			raise ArgumentException, "There are only #{available_rooms.length} available for the dates selected"
+		else
+			rooms = []
+			number_of_rooms.times do |i|
+				rooms << available_rooms[i]
+				@rooms_currently_in_block << available_rooms[i]
+			end
+			new_block = Hotels::Block_Of_Rooms.new(rooms, date, id)
+			@room_blocks << new_block
+		end
+	  end
+	  
+	  #just looks for at least 1 avaialble room in a given block
+	  def check_block_for_availablity(block_id)
+		@room_blocks.each do |block|
+			if block.id == block_id
+				block.room_booked.each do |is_room_booked|
+					if is_room_booked == false
+						return true
+					end
+				end
+			end
+		end
+		
+		return false
+	  end
+	  
+	  def reserve_room_from_block(block_id)
+		if @room_blocks == []
+			raise ArgumentError, "There are no current room blocks"
+		else
+			@room_blocks.each do |current_block|
+				if current_block.id == block_id
+					if check_block_for_availablity(block_id)
+						room = get_open_room_from_block(current_block)
+						new_reservation = Hotels::Reservation.new(room, current_block.date, 175)
+					else
+						puts "There are no rooms avaiable for this block"
```

Comment:
This should be raising an error of some kind (probably a custom one), instead of printing to the screen with `puts`.